### ISSUE TITLE
Java : Canceling multiple tasks in a single vm/host 

### ISSUE THREAD

**Issue by handaoui-mohamed** [simgrid/simgrid#328]
I'm running multiple tasks in one vm, but sometime I need to cancel some tasks to reduce the host load, here is an example :
I'm using Simgrid **v3.23**
```java
List<LoadRunner> loadRunners = new ArrayList<>();
for (int i = 0; i < 10; i++) {
    LoadRunner loadRunner = new LoadRunner(vm, "Task-" + i, 1.84E10);
    loadRunner.start();
    loadRunners.add(loadRunner);
}

for (int i = 0; i < 10; i++) {
    loadRunners.get(i).cancel();
}
```

LoadRunner class:
```java
public class LoadRunner extends Process {
...
@Override
    public void main(String[] strings) throws MsgException {
        try {
            task = new Task("TASK" + getName(), flopsAmount, 0);
            started = true;
            task.execute();

            double duration = Msg.getClock() - start;
            Msg.info("Duration " + getName() + " " + Msg.getClock());
        } catch (TaskCancelledException e) {
            System.out.println(Msg.getClock() + " Task " + getName() + " is canceled");
            task = null;
        }
    }

    public void cancel() {
        System.out.println(Msg.getClock() + " Canceling " + getName());
        task1.cancel();
    }
...
}
```

This is the output of the log, unfortunatly it does not say much, it seems to be a problem with the host load update function:
```
[0.000000] /home/mhandaoui/tools/SimGrid-3.23/src/kernel/actor/ActorImpl.cpp:461: [simix_process/VERBOSE] Create context Test-0
0.0 Canceling Test-0
[0.000000] /home/mhandaoui/tools/SimGrid-3.23/src/kernel/activity/ActivityImpl.cpp:60: [simix_process/VERBOSE] Activity 0x7f3328002bd0 is canceled
0.0 Task Test-0 is canceled
[0.000000] /home/mhandaoui/tools/SimGrid-3.23/src/kernel/actor/ActorImpl.cpp:461: [simix_process/VERBOSE] Create context Test-1
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f3387c8552b, pid=16372, tid=0x00007f33ad89c700
#
# JRE version: Java(TM) SE Runtime Environment (8.0_212-b10) (build 1.8.0_212-b10)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.212-b10 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libsimgrid.so+0x27c52b]  simgrid::plugin::HostLoad::update()+0x7b
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /home/mhandaoui/projets/poc-hadoop-simulator/hs_err_pid16372.log
#
```

I can call `host.getAvgLoad` or `host.getCurrentLoad` with no problem since the plugin is  initialized `Msg.loadInit()`. If I don't initialize the host load plugin the previous code works fine. 

**This is a working example: 
[simgrid-test.zip](https://github.com/simgrid/simgrid/files/3363006/simgrid-test.zip)**

**Comment by frs69wq** [comment]
Hi, 
Could you please add a minimal working example that reproduces this bug to the issue? It would greatly help us to track down the origin of the problem and propose a solution.
Thanks,
F.

**Comment by handaoui-mohamed** [comment]
> Hi,
> Could you please add a minimal working example that reproduces this bug to the issue? It would greatly help us to track down the origin of the problem and propose a solution.
> Thanks,
> F.

Sorry, I attached a working project that reproduces this bug, I hope this would be helpful.

**Comment by handaoui-mohamed** [comment]
I found the problem in `void HostLoad::update()` where `activity->surf_action_` is sometime NULL for some reason. However I don't know whether it is the main reason for my issue.
For the moment I'm just checking for the NULL reference to avoid the error.
```c++
auto action =  activity->surf_action_;
XBT_DEBUG("\n**Action is null = %d\n",(action == NULL));

if (action != NULL && action->get_finish_time() != now && activity->state_ == e_smx_state_t::SIMIX_RUNNING) {
  if (remaining_cost_after_last_update == activity_uninitialized_remaining_cost) {
    remaining_cost_after_last_update = action->get_cost();
  }
  double computed_flops_since_last_update = remaining_cost_after_last_update;
  computed_flops_                        += computed_flops_since_last_update - activity->get_remaining();
  remaining_cost_after_last_update        = activity->get_remaining();
}
else if (activity->state_ == e_smx_state_t::SIMIX_DONE) {
  computed_flops_ += remaining_cost_after_last_update;
  current_activities.erase(current_iter);
}
```

**Comment by mquinson** [comment]
Ok, I applied your fix since it sounds reasonable and does not break any test. Next time if you propose a PR, I'll merge your first contribution to SimGrid!

Thanks for your time,
Mt
